soti-stm32project-boardV1/Core/Src/main.c

```diff
@@ -396,60 +399,59 @@ static void MX_GPIO_Init(void)
 }
 
 /* USER CODE BEGIN 4 */
-/**
-  * @brief  Rx Fifo 0 message pending callback
-  * @param  hcan: pointer to a CAN_HandleTypeDef structure that contains
-  *         the configuration information for the specified CAN.
-  * @retval None
-  */
-void HAL_CAN_RxFifo0MsgPendingCallback(CAN_HandleTypeDef *hcan1)
+void on_message_received(CANMessage msg)
 {
-  HAL_StatusTypeDef operation_status;
-  operation_status = CAN_Message_Received();
-  if (operation_status != HAL_OK)
-  {
-    char *str1 = "ERROR:";
-    char *str2 = "CAN RX";
-    LCD_CLEAR_DISPLAY();
-    LCD_PRINT_STR(str1, 0);
-    LCD_PRINT_STR(str2, 16);
-  }
+	CANMessage message = msg;
+	uint8_t serializedData[11];
+
+	//serializing the CANMessage to transfer over UART.
+	serializeCANMessage(&message, serializedData);
+	//transferring data over UART.
+	HAL_UART_Transmit(&huart3, serializedData, sizeof(serializedData), HAL_MAX_DELAY);
 }
 
-void HAL_UART_RxCpltCallback(UART_HandleTypeDef *huart)
+void on_error_occured(CANWrapper_ErrorInfo error)
 {
-  CANMessage_t message;
+	char *str1 = "ERROR:";
+	char *str2 = "CAN RX";
+	LCD_CLEAR_DISPLAY();
+	LCD_PRINT_STR(str1, 0);
+	LCD_PRINT_STR(str2, 16);
+}
 
+void HAL_UART_RxCpltCallback(UART_HandleTypeDef *huart)
+{
+	CANMessage message;
   deserializeCANMessage(&message, canRxData);
 
-  CAN_Queue_Enqueue(&groundToSatelliteQueue, &message);
+  CANQueue_Enqueue(&groundToSatelliteQueue, (CANQueueItem){ .msg = message });
 
   HAL_UART_Receive_IT(&huart3, canRxData, sizeof(canRxData));
 }
 
-void serializeCANMessage(CANMessage_t* message, uint8_t* serializedData)
+void serializeCANMessage(CANMessage* message, uint8_t* serializedData)
 {
   serializedData[0] = message->priority;
-  serializedData[1] = message->SenderID;
-  serializedData[2] = message->DestinationID;
-  serializedData[3] = message->command;
+  serializedData[1] = message->sender;
+  serializedData[2] = message->recipient;
+  serializedData[3] = message->cmd;
 
   for (int i = 0; i < 7; i++)
```

Comment:
```suggestion
  for (int i = 0; i < CAN_MAX_BODY_SIZE; i++)
```

Same for the other loop